version: '3.8'

services:
  tasks-service:
    build: .
    command: python /app/main.py
    env_file:
      - .env
    depends_on:
      user-management-service:
        condition: service_started
      db:
        condition: service_healthy
      manager:
        condition: service_started
    volumes:
      - ./app:/app
    ports:
      - ${TASKS_SERVICE_PORT}:${TASKS_SERVICE_PORT}
    networks:
      - microservice-network

  user-management-service:
    image: ghcr.io/projekt-datascience/backend-user-management:release
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy
      manager:
        condition: service_started
    ports:
      - ${USER_MANAGEMENT_SERVICE_PORT}:${USER_MANAGEMENT_SERVICE_PORT}
    networks:
      - microservice-network

  db:
    image: postgres:latest
    restart: always
    env_file:
      - .env
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
      PGPORT: ${DB_PORT}
    ports:
      - ${DB_PORT}:${DB_PORT}
    networks:
      - microservice-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready"]
      interval: 10s
      timeout: 5s
      retries: 5


  db-manager:
    image: dpage/pgadmin4:latest
    env_file:
      - .env
    restart: always
    depends_on:
      db:
        condition: service_healthy
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_PASSWORD}
    ports:
      - "8080:80"
    networks:
      - microservice-network


  manager:
    image: ghcr.io/projekt-datascience/manager:release
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - manager_volume:/root
    networks:
      - microservice-network

networks:
  microservice-network:

volumes:
  manager_volume: